src/drug_discovery/abfe.py

```diff
@@ -0,0 +1,134 @@
+"""This module encapsulates methods to run ABFE and show ABFE results on Deep Origin.
+
+The ABFE object instantiated here is contained in the Complex class is meant to be used within that class."""
+
+from typing import Optional
+
+import pandas as pd
+from beartype import beartype
+from deeporigin.data_hub import api
+from deeporigin.drug_discovery import chemistry as chem
+from deeporigin.drug_discovery import utils
+from deeporigin.exceptions import DeepOriginException
+from deeporigin.utils.core import PrettyDict
+
+
+class ABFE:
+    """class to handle ABFE-related tasks within the Complex class.
+
+    Objects instantiated here are meant to be used within the Complex class."""
+
+    def __init__(self, parent):
+        self.parent = parent
```

Comment:
@laionazeredo note: i've made parent a required param on initialization. so it can never be None